.github/workflows/ci.yml

```diff
@@ -191,88 +187,90 @@ jobs:
-          destination: gh-zcash/${{ github.run_id }}/${{ matrix.name }}/src
 
       - name: Upload zcashd-wallet-tool artifact
-        uses: google-github-actions/upload-cloud-storage@v2
+        uses: actions/upload-artifact@v4
         with:
+          name: ${{ format('./src/zcashd-wallet-tool{0}', matrix.file_ext) }}-${{ matrix.name }}
           path: ${{ format('./src/zcashd-wallet-tool{0}', matrix.file_ext) }}
-          destination: gh-zcash/${{ github.run_id }}/${{ matrix.name }}/src
 
       - name: Upload zcash-inspect artifact
-        uses: google-github-actions/upload-cloud-storage@v2
+        uses: actions/upload-artifact@v4
         with:
+          name: ${{ format('./src/zcash-inspect{0}', matrix.file_ext) }}-${{ matrix.name }}
+          path: ${{ format('./src/zcash-inspect{0}', matrix.file_ext) }}
+
+
+      - name: Upload zcash-inspect artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: ${{ format('./src/zcash-inspect{0}', matrix.file_ext) }}-${{ matrix.name }}
           path: ${{ format('./src/zcash-inspect{0}', matrix.file_ext) }}
-          destination: gh-zcash/${{ github.run_id }}/${{ matrix.name }}/src
 
       - name: Upload zcash-tx artifact
-        uses: google-github-actions/upload-cloud-storage@v2
+        uses: actions/upload-artifact@v4
         with:
+          name: ${{ format('./src/zcash-tx{0}', matrix.file_ext) }}-${{ matrix.name }}
           path: ${{ format('./src/zcash-tx{0}', matrix.file_ext) }}
-          destination: gh-zcash/${{ github.run_id }}/${{ matrix.name }}/src
 
       - name: Upload zcash-btest artifact
-        uses: google-github-actions/upload-cloud-storage@v2
+        uses: actions/upload-artifact@v4
         with:
+          name: ${{ format('./src/test/test_bitcoin{0}', matrix.file_ext) }}-${{ matrix.name }}
           path: ${{ format('./src/test/test_bitcoin{0}', matrix.file_ext) }}
-          destination: gh-zcash/${{ github.run_id }}/${{ matrix.name }}/src/test
 
       - name: Upload zcash-gtest artifact
-        uses: google-github-actions/upload-cloud-storage@v2
+        uses: actions/upload-artifact@v4
         with:
+          name: ${{ format('./src/zcash-gtest{0}', matrix.file_ext) }}-${{ matrix.name }}
           path: ${{ format('./src/zcash-gtest{0}', matrix.file_ext) }}
-          destination: gh-zcash/${{ github.run_id }}/${{ matrix.name }}/src
 
       - name: Upload src/test/buildenv.py
-        uses: google-github-actions/upload-cloud-storage@v2
+        uses: actions/upload-artifact@v4
         with:
+          name: ./src/test/buildenv.py-${{ matrix.name }}
           path: ./src/test/buildenv.py
-          destination: gh-zcash/${{ github.run_id }}/${{ matrix.name }}/src/test
 
       - name: Upload src/secp256k1
-        uses: google-github-actions/upload-cloud-storage@v2
+        uses: actions/upload-artifact@v4
         with:
+          name: ./src/secp256k1-${{ matrix.name }}
           path: ./src/secp256k1
```

Comment:
This includes both source and executables. Is that what we want, or do we want just what is needed to run the tests?

Reply:
Include both, since the tests need the executables.